aidants_connect_carto/apps/core/search.py

```diff
@@ -91,15 +134,23 @@ def _build_results(self):
         to the calling code, typically as a more or less complex `dict`, which
         could also contain meta-information about the search, like the total
         number of results, ...
+        - order_by
+        - opening_hours
+        - pagination
         """
-        page_number = self.query.get("page", 1)
-        offset = self.RESULTS_PER_PAGE * (page_number - 1)
+        # order_by
+        order_by_field = self.query.get("order_by", self.DEFAULT_ORDER_BY)
+        self.queryset = self.queryset.order_by(order_by_field)
+
+        raw_results = self.queryset
 
-        raw_results = self.queryset[
-            offset : offset + self.RESULTS_PER_PAGE
-        ]  # <-- db query is executed here!
+        # opening_hours (place.is_open) filter
+        if self.query.get("opening_hours", "") != "":
+            raw_results = [place for place in raw_results if place.is_open]
 
-        # We could now build a refined representation of our results,
-        # but right now this is merely a proof of concept, so we just...
+        # paginator <-- db query is executed here!
+        paginator = Paginator(raw_results, self.RESULTS_PER_PAGE)
```

Comment:
Le même genre de considérations de performance peut s'appliquer à terme à la pagination, et selon la manière spécifique dont ton code se déroule, il se peut que la requête SQL ait déjà été exécutée à ce stade (typiquement, dans notre cas, Django aura été obligé de l'exécuter plus haut pour évaluer `place.is_open`) auquel cas on se retrouve avec potentiellement _des milliers de lignes_ ayant circulé sur le réseau, pour finalement en jeter une immense majorité pour ne garder que celles de la _page_ qui nous intéresse :)

Là encore, c'est _good enough for now_, mais tu peux avoir en tête que ça risque de ne pas super bien _scaler_ en fonction de la volumétrie.

Reply:
cf le commentaire précédent :)